pennylane/devices/preprocess.py

```diff
@@ -136,46 +139,57 @@ def validate_device_wires(
             f"Abstract wires are not yet supported."
         )
 
-    if wires:
+    if not wires:
+        return (tape,), null_postprocessing
 
-        if any(qml.math.is_abstract(w) for w in wires):
-            raise WireError(
-                f"Cannot run circuit(s) on {name} as abstract wires are present in the device: {wires}. "
-                f"Abstract wires are not yet supported."
-            )
+    if any(qml.math.is_abstract(w) for w in wires):
+        raise WireError(
+            f"Cannot run circuit(s) on {name} as abstract wires are present in the device: {wires}. "
+            f"Abstract wires are not yet supported."
+        )
 
-        if extra_wires := set(tape.wires) - set(wires):
-            raise WireError(
-                f"Cannot run circuit(s) on {name} as they contain wires "
-                f"not found on the device: {extra_wires}"
-            )
+    if extra_wires := tuple(
+        w for w in set(tape.wires) - set(wires) if not isinstance(w, DynamicWire)
+    ):
+        raise WireError(
+            f"Cannot run circuit(s) on {name} as they contain wires "
+            f"not found on the device: {extra_wires}"
+        )
 
-        modified = False
-        new_ops = None
-        for i, op in enumerate(tape.operations):
-            if isinstance(op, qml.Snapshot):
-                mp = op.hyperparameters["measurement"]
-                if not mp.wires:
-                    if not new_ops:
-                        new_ops = list(tape.operations)
-                    modified = True
-                    new_mp = copy(mp)
-                    new_mp._wires = wires  # pylint:disable=protected-access
-                    new_ops[i] = qml.Snapshot(
-                        measurement=new_mp, tag=op.tag, shots=op.hyperparameters["shots"]
-                    )
-        if not new_ops:
-            new_ops = tape.operations  # no copy in this case
+    if any(isinstance(w, DynamicWire) for w in tape.wires) and any(
+        isinstance(mp, StateMP) for mp in tape.measurements
+    ):
+        raise DeviceError(
+            "returning the state is currently not supported with dynamic wires. "
+            " Consider using the density_matrix or probabilities instead."
```

Comment:
As we are not referring to software implementations (`probs`) but to the proper names ("probabilities") for the latter, I'd also do so for the former :) 
```suggestion
            " Consider using the density matrix or probabilities instead."
```

Reply:
Realized I placed this there when I was thinking about this a different order. moving it to `device_resolve_dynamic_wires`.